### DISCUSSION TITLE
implementing mouse interactions on shapes

### DISCUSSION THREAD

**Discussion by snvgglebear** [fyne-io/fyne#3300] [Q&A] [answered]
How would I use dragevents, pointevents, mouse events, and other mouse interactions on shapes? (Lines, rectangles, etc.)

**Comment by Jacalz** [comment, marked as answer]
You can make sure to wrap the centavos objects in a custom widget and then implement the interfaces for supporting the given mouse movement (`desktop.Mouseable`, `desktop.Hoverable`., `fyne.Draggable` and so on).

I believe that most of what you ask about already has been mentioned here:
https://github.com/fyne-io/fyne/discussions/3261

**Comment by lordofscripts** [comment]
What about the mouse wheel that gets rolled forward or backward which is typically used for scrolling or up/down values? I don't see any reference to that.

**Reply by andydotxyz** [reply]
Please don't comment on old, closed discussions to ask a new question.
You are looking for fyne.Scrollable.
